index.rst

```diff
@@ -1,59 +1,367 @@
+
+In cases where one or more private ``RUN`` collections contain datasets that should not be considered part of the final public outputs (e.g. because they are superceded by datasets in other private ``RUN`` collections), a ``TAGGED`` collection can be used to screen and aggregate these.
+That ``TAGGED`` collection would then be a direct child of the final public ``CHAINED`` collection, instead of any ``RUN`` collections it references.
+
+.. note::
+
+   It is not generally possible to use a ``TAGGED`` collection as the public output collection for a processing run, because putting master calibrations (which are almost always inputs, even if indirectly) in a ``TAGGED`` collection strips them of their validity ranges and does not allow datasets from different validity ranges to coexist.
+   So even if a ``TAGGED`` collection is used, the public ``CHAINED`` collection would contain both that collection and the input ``CALIBRATION`` collection as children.
+
+.. note::
+
+   These public ``CHAINED`` collections essentially mimic Gen2's "parent link" mechanism, which provides at best approximate coarse-grained provenance information about which datasets were used as inputs when producing others.
+   The Gen3 repository will eventually be extended to include fine-grained, exact provenance - essentially a serialization of the directed acyclic graph (DAG) that describes the processing.
+   Whether queries against that DAG are fast enough to allow this more rigorous provenance information to be used as a type of collection (replacing some usage of ``TAGGED`` and ``CHAINED`` collections) remains to be seen, however.
+   It is also worth noting that in general the full DAG does not maintain the usual collection invariant of having only one dataset with a particular dataset type and data ID (e.g. two calexps with the same data ID, from two differently-configured runs, could each contribute to different, non-conflicting coadd patches in downstream runs).
+
+.. _collections-developer-processing-outputs:
+
+Developer processing outputs
+----------------------------
+
+Processing initiated by DM developers that are intended primarily for personal or small-group use must start with ``u/<user>`` (e.g. ``u/jbosch``), and are strongly encouraged to start with ``u/<user>/<ticket>`` (e.g. ``u/jbosch/DM-56789``) whenever possible.
+Names and structure after this prefix are at user discretion, but we strongly recommend using a combination of ``CHAINED`` collections and ``RUN`` collections to distinguish between "inputs and outputs" collections and "output only" collections, as in :ref:`collections-shared-official-processing-outputs`.
+The ``pipetask`` tool will automatically take care of this if the ``--output`` option is used with or instead of the ``--output-run`` option.
+
```

Comment:
Can we ask for clarification on this RFC PR too? These two sentences seem important but went over my head. Example?

Reply:
If I run
```
pipetask run -b /datasets/repo --input HSC/defaults --output u/jbosch/DM-27147/test 
```
the collections I get will look like this (this option to `query-collections` is new):
```
$ butler query-collections /datasets/repo --chains=tree u/jbosch/DM-27147/test
                 Name                        Type
------------------------------------------- -----------
u/jbosch/DM-27147/test                      CHAINED
  u/jbosch/DM-27147/test/20201116T09h28m08s RUN
  HSC/defaults                              CHAINED
    HSC/raw/all                             RUN
    HSC/masks/S18A                          RUN
    refcats                                 CHAINED
      refcats/gen2                          RUN
    HSC/calib                               CHAINED
      HSC/calib/gen2/defaults               CALIBRATION
    skymaps                                 RUN
    HSC/masks                               CHAINED
      HSC/masks/S18A                        RUN
```
Indentation indicates membership in the CHAINED collection in the previous line.  The collection I specified with `--output` is the CHAINED collection at the top, and the first entry in that is a RUN collection with a timestamp-suffixed name that contains the actual processing outputs of that run.  The top-level CHAINED collection also contains `HSC/defaults`, the input collection I specified, and that recursively contains many other collections.

If I'd passed `--output-run u/jbosch/DM-27147/test/a` as well, that name would be used instead of `u/jbosch/DM-27147/test/20201116T09h28m08s`, but the content and structure would be the same.

In either case, one can initialize a butler with
```
butler = Butler("/datasets/repo", collections="u/jbosch/DM-27147/test")
```
and use it to `get` _both_ outputs and inputs, just as in Gen2.

But If one passes `--output-run` _without_ `--output`, the top-level CHAINED collection is not created, and there's no single collection that can be used to `get` outputs and inputs.

Reply:
Oh, an example like this in the text would be very useful, please!